core/src/main/scala/org/apache/spark/deploy/rest/RestSubmissionClient.scala

```diff
@@ -277,8 +277,8 @@ private[spark] class RestSubmissionClient(master: String) extends Logging {
   private def getBaseUrl(master: String): String = {
     var masterUrl = master
     supportedMasterPrefixes.foreach { prefix =>
-      if (master.startsWith(prefix)) {
-        masterUrl = master.stripPrefix(prefix)
+      if (masterUrl.startsWith(prefix)) {
+        masterUrl = masterUrl.stripPrefix(prefix)
```

Comment:
> How do I then get a or against master into 1.4 branch?
Just check out the master branch and implement your changes there, then we'll back port it into branch-1.4 when we merge.